### PR TITLE
Videos show

### PR DESCRIPTION
**Pull request by tomohung**

### CODE REVIEW

File: app/controllers/reviews_controller.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      redirect_to @video
+    else
+      @reviews = @video.reviews.reload
+      render 'videos/show'
+    end
+  end
+
+  private
+
+  def set_params
```
Comment (brandoncc) [inline app/controllers/reviews_controller.rb]: This method name is misleading. It sounds like you are changing the params. The general practice in naming strong param methods is `model_params`. In this case that would be `review_params`.

File: app/views/shared/_header.html.haml [hunk 2]
```diff
@@ -22,4 +22,4 @@
             %li
               %a(href="#") Account
               %a(href="#") Plan and Billing
-              = link_to 'Sign Out', sign_out_path
+              = link_to 'Sign Out', sign_out_path, method: :delete
```
Comment (brandoncc) [inline app/views/shared/_header.html.haml:25]: You may run into an issue on Heroku with this. I just dealt with this with another student and for some reason Heroku refused to use anything other than a `GET` route in his app. I am not sure if there was an outside factor as it should have worked fine.

File: app/views/videos/show.html.haml [hunk 3]
```diff
@@ -15,35 +15,31 @@
...
+                
           .form-group
             %label Write Review
             .row
               .col-sm-8
-                %textarea.form-control(name="" rows="6")
-        %fieldset.form-group.actions.clearfix
-          %input(type="submit" value="Submit" class="btn")
-          %a(href="") Cancel
+                =f.text_area :context, row: 6, class: 'span6'
```
Comment (brandoncc) [inline app/views/videos/show.html.haml]: The classes in Bootstrap 3 are like this `col-xs-6`. `span6` is no longer around.

File: app/views/shared/_header.html.haml (hunk 2 above)
Comment (brandoncc) [inline app/views/shared/_header.html.haml:25]: Never mind, I just pushed your app to Heroku and it works fine. There must be an outside factor on his app.

File: spec/controllers/review_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+      end
+
+      context 'with valid input' do
+
+        before do
+          post :create, review: Fabricate.attributes_for(:review), video_id: video.id
+        end
+        
+        it 'should sets @video' do
+          expect(Video.first).to eq(video)
```
Comment (brandoncc) [inline spec/controllers/review_controller_spec.rb:23]: This spec is flawed. You are fabricating this video on line 6, that means that it already exists in the db. This is not testing your create action. The proper way to test that would be to manually set a title in the fabricator on line 19 and then test for the title on the first video. By the way, this is an example of why you have to be careful with using let and before. The fabrication on line 6 was so far away that you didn't realize that it was affecting your spec. It is better to have all of the code your spec relies on in your spec, even if that means duplication.

Comment (brandoncc) [inline spec/controllers/review_controller_spec.rb:23]: Okay, I am apparently tired. I apologize. I didn't realize that this spec was testing the creation of a review, not a video. Ignore my last comment...the last part of it is still valid though.

File: spec/controllers/sessions_controller_spec.rb [hunk 5]
```diff
@@ -18,9 +18,8 @@
 
   describe 'POST create' do
     context 'with authenticated' do
-      
-      let(:user) { Fabricate(:user) } 
-      before do   
+      before do 
+        user = Fabricate(:user)
```
Comment (brandoncc) [inline spec/controllers/sessions_controller_spec.rb]: You had it right before. You don't have access to `user` when it is in the `before` block. You should change this spec back to how it was.

File: app/views/videos/show.html.haml (hunk 3 above)
Comment (tomohung) [inline app/views/videos/show.html.haml]: This gives me a big hand!! That's why my template always looks weird.

Comment (brandoncc) [inline app/views/videos/show.html.haml]: You can review the new css classes here http://getbootstrap.com/css/#grid
The classes that you were using are from Bootstrap 2.3.2, which you used in course 2.

**Pull request merged**